beacon-chain/core/helpers/signing_root.go

```diff
@@ -98,8 +98,8 @@ func VerifySigningRoot(obj fssz.HashRoot, pub, signature, domain []byte) error {
 }
 
 // VerifyBlockSigningRoot verifies the signing root of a block given it's public key, signature and domain.
-func VerifyBlockSigningRoot(blk *ethpb.BeaconBlock, pub, signature, domain []byte) error {
-	set, err := BlockSignatureSet(blk, pub, signature, domain)
+func VerifyBlockSigningRoot(rootFunc func() ([32]byte, error), pub, signature, domain []byte) error {
```

Comment:
While on it, probably it is worth moving `rootFunc` argument to the end of argument list (for consistency, similar to `BlockSignatureSet()`)?